### ISSUE TITLE
Yarn cannot resolve relative paths in .yarnrc correctly

### ISSUE THREAD

**Issue by dereli** [yarnpkg/yarn#6057]
**Do you want to request a *feature* or report a *bug*?** Bug

**What is the current behavior?**
If *cafile* variable in the .yarnrc file is relative, the path to file is resolved relative to current working directory, instead it should be relative to the path of the .yarnrc file.

**If the current behavior is a bug, please provide the steps to reproduce.**

https://gist.github.com/dereli/948095221b45fa2e623070a03b1184bb

**What is the expected behavior?**
Yarn should be able to find correct CAFile resolving the path relative to .yarnrc.

**Please mention your node.js, yarn and operating system version.**
$ yarn -v
1.7.0
$ node -v
v8.9.4
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.5
BuildVersion:	17F77

**Comment by dereli** [comment]
@arcanis thanks for the information.

I spent some time on this, but those changes don't solve the problem immediately.

As you suggested, I added the `cafile` key to the lists. This triggers a `path.resolve` with correct params; however, just right after path resolution, it tries to `fs.mkdirp` and it fails with an `EEXIST` error. This makes sense for the rest of the options but not for the `cafile` option.
